app/frontend/src/features/messages/groupchats/groupChatsActions.ts

```diff
@@ -0,0 +1,74 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { User } from "../../../pb/api_pb";
+import { GroupChat } from "../../../pb/conversations_pb";
+import { RootState } from "../../../reducers";
+import { service } from "../../../service";
+import { groupChatsFetched, messagesFetched, groupChatSet } from ".";
+
+type FetchGroupChatsArguments = void;
+
+export const fetchGroupChats = createAsyncThunk<
+  void,
+  FetchGroupChatsArguments,
+  { state: RootState }
+>("hostRequests/fetchGroupChats", async (_, thunkApi) => {
+  const groupChats = await service.conversations.listGroupChats();
+  thunkApi.dispatch(groupChatsFetched(groupChats));
+});
```

Comment:
```suggestion
export const fetchGroupChats = createAsyncThunk("hostRequests/fetchGroupChats", async () => {
  return service.conversations.listGroupChats();
});
```
The explicit typing on state is unnecessary as you are not retrieving state in this function, so you can simplify and let the underlying typing infer the types.

Also, since dispatching an action created from `createAsyncThunk` already auto dispatch actions for you when it resolves/rejects, it'll be simpler to return the result and set the `groupChats` state in `fetchGroupChats.fulfilled` in the slice. This is just adding extra noise for no apparent benefits and it's the kinda thing you'd do only if you were writing things by hand and when you are not using Redux toolkit.

Reply:
This is what I meant when I said it would be better to get rid of setMessages. I missed this last night must've been looking at the wrong diff, sorry!